setup.py

```diff
@@ -30,6 +31,19 @@
 '''.lstrip()
 
 
+def cache_version_id():
+    """Returns the version id to use for the incremental hash.
+
+    If setup.py is run from a git repo, the git commit hash will be
+    included if possible. If not, then this function will fall back to
+    using the default version id from mypy/version.py."""
+    if git.is_git_repo('.') and git.have_git():
```

Comment:
I kind of worry that pip install might not have the current directory equal to the repo. I'll try to play with this more to determine if this is a problem or not.